### ISSUE TITLE
Thread syntax sugar

### ISSUE THREAD

**Issue by aleclarson** [jamiebuilds/ghost-lang#7]
It might be cool if there was syntax sugar for "threads" (the definition of which depends on the target platform). For example, threads would compile to web workers for a browser target.

**Comment by jamiebuilds** [comment]
Ghost has reached a point where it would have to be compiled to Wasm which is starting to have support for threads now https://developers.google.com/web/updates/2018/10/wasm-threads

I'm not sure what's needed for syntax other than potentially a syntax for moving/copying values into closures to be spawned as a thread

**Comment by SCKelemen** [comment]
This looks a lot like a language I have been working on. Have you consider alternative concurrency models such as Actors or CSP?

**Comment by jamiebuilds** [comment]
Actors, CSP, etc are things you would implement on top of threads. I think they would be more in the stdlib.

**Comment by aleclarson** [comment]
Thoughts on cooperative threading vs preemptive threading?

**Comment by SCKelemen** [comment]
I don't think that needs to be true. CSP, Actors, and threads are all independent concurrency paradigms. All of them can be implemented completely independent and without any other, and each can be implemented on any other.
